File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int[] newNums = new int[nums.length];
        int visited = -1;
        int k = 0;

        for(int i = 0; i < nums.length; i++) {
            for(int j = i; j < nums.length; j++) {
                if(j > visited) {
                    if(nums[j] != val) {
                        visited = j;
                        newNums[i] = nums[j];
                        k++;
                        break;
                    }
                    else {
                        visited = j;
                    }
                }
            }
        }

        for(int i = 0; i < newNums.length; i++) {
            nums[i] = newNums[i];
        }

        return k;
    }
}

File: 0027-remove-element/0027-remove-element.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut k: usize = 0;

        for i in 0..nums.len() {
            if nums[i] != val {
                nums[k] = nums[i];
                k += 1;
            }
        }

        k.to_string().trim().parse::<i32>().unwrap()
    }
}